app/views/spree/admin/product_properties/index.html.haml

```diff
@@ -0,0 +1,72 @@
+= render partial: 'spree/admin/shared/product_sub_menu'
+
+= render partial: 'spree/admin/shared/product_tabs', locals: { current: 'Product Properties' }
+
+= render partial: 'spree/shared/error_messages', locals: { target: @product }
+
+- content_for :page_actions do
+  %ul.tollbar.inline-menu
+    %li
+      = link_to_add_fields Spree.t(:add_product_properties), 'tbody#product_properties', class: 'icon-plus button'
+    %li
+      %span#new_ptype_link
+        = link_to Spree.t(:select_from_prototype), available_admin_prototypes_url, remote: true, 'data-update' => 'prototypes', class: 'button icon-copy'
+
+= form_for @product, url: admin_product_url(@product), method: :put do |f|
+  %fieldset.no-border-top
+    .add_product_properties{"data-hook" => "add_product_properties"}
+    #prototypes{"data-hook" => ""}
+    = image_tag 'select2-spinner.gif', plugin: 'spree', style: 'display:none;', id: 'busy_indicator'
+
+    %table.index.sortable{"data-hook" => "", "data-sortable-link" => update_positions_admin_product_product_properties_url}
+      %thead
+        %tr{"data-hook" => "product_properties_header"}
+          %th.no-border
+          %th= t('admin.products.properties.property_name')
+          %th= t('admin.description')
+          %th.actions
+
+      %tbody#product_properties{"data-hook" => ""}
+        = f.fields_for :product_properties do |pp_form|
+          = render partial: 'product_property_fields', locals: { f: pp_form }
+
+    = f.check_box :inherits_properties
+    = f.label :inherits_properties, t(".inherits_properties_checkbox_hint", supplier: @product.supplier.name)
+    %br
+    %br
+
+    #inherited_properties
+      %table.index
+        %thead
+          %tr{"data-hook" => "producer_properties_header"}
+            %th= t('admin.products.properties.inherited_property')
+            %th= t('admin.description')
+            %th.actions
+        %tbody#producer_properties{"data-hook" => ""}
+          - @product.supplier.producer_properties.each do |producer_property|
+            %tr
+              %td= producer_property.property.presentation
+              %td= producer_property.value
+              %td.actions
+
+
+    = render partial: 'spree/admin/shared/edit_resource_links'
+
+    = hidden_field_tag 'clear_product_properties', 'true'
+
+:coffee
+  $(document).ready ->
+    $("#inherited_properties").toggle $("input#product_inherits_properties").is(':checked')
+  $("input#product_inherits_properties").change ->
+    $("#inherited_properties").toggle $(this).is(':checked')
+
+= javascript_tag do
+  var properties = #{raw(@properties.to_json)};
```

Comment:
there's an indentation error here.
I found these js convertions from erb to haml to break easily. I dont think this raw and to_json will work like this...
Have you tested this? Give it a try. If it's broken and it's hard to fix, maybe it's acceptable to delete this autocomplete function?

Reply:
I had errors when trying to use correct indentation with `= javascript_tag`, so I changed it to `:javascript`. No errors anymore, and the autocomplete works fine !